src/transformStateWithClones.js

```diff
@@ -7,7 +7,40 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateCopy = { ...state };
+  const resultState = [];
+
+  for (const action of actions) {
+    switch (true) {
+      case (action.type === 'addProperties'):
+        resultState.push(Object.assign({}, stateCopy, action.extraData));
+        Object.assign(stateCopy, action.extraData);
```

Comment:
Checklist p.4 [DONT REPEAT YOURSELF] - If you perform same action in all switch cases - do it just once afterwards.
So, basically, just do your .push() once in the end of your switch, in the body of for of loop like that: 
resultState.push({ ...stateCopy });
<sub>Same problem with the names of your variables, but I think that's up to you</sub>